src/main/cli/process.ts

```diff
@@ -0,0 +1,70 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app } from "electron";
+import { ICli, ICliParam } from "readium-desktop/main/cli/commandLine";
+import { Arguments } from "yargs";
+
+declare const __APP_NAME__: string;
+declare const __APP_VERSION__: string;
+
+// Logger
+const debug = debug_("readium-desktop:cli");
+
+export function printHelp(commandLine: ICli[], argv: Arguments) {
+
+    if (argv.help || argv.h) {
+        process.stdout.write(`${__APP_NAME__} ${__APP_VERSION__}\n`);
+        process.stdout.write("\n");
+        process.stdout.write(`Usage: ${argv.$0} [options][paths...]\n`);
+        process.stdout.write("\n");
+        process.stdout.write(`Options\n`);
+        commandLine.map((v) => {
+            if (v.help.length) {
+                process.stdout.write(` ${v.help[0]}\t\t${v.help[1] ? v.help[1] : ""}\n`);
+            }
+        });
+        app.exit(0);
+        process.exit(0);
+    }
+}
+
+export async function processCommandLine(commandLine: ICli[], argv: Arguments): Promise<boolean> {
+
+    const param: ICliParam = {
+        argv,
+        quit: false,
+    };
+    debug("param: ", param);
+    let returnCode = 0;
+
+    // assign arg and remove $0
+    const arg = Object.keys(argv);
+    arg.splice(arg.indexOf("$0"), 1);
+
+    // execute all commands
+    const commandPromiseTab = arg.map((command) => {
```

Comment:
Some items inside `commandPromiseTab` might be `undefined`, because not all code paths inside the iterated function return a value.
I think the intent of this code routine is to "filter" (not to return a one-to-one "map"), therefore I would suggest using `Array.filter()` with an inclusion predicate, instead of `Array.map()`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
(this will eliminate the ambiguity for future readers of the source code)

Reply:
filter function returns a string Array and not a Promise<boolean>.
And the undefined is handle in Promise.all()

Reply:
Sure, then you chain filter().map().map() (fluent design, chained functions).
Using `.map()` to return `undefined` values in the returned array is an **anti-pattern**, the code is misleading because the assumption is that there is a one-to-one "mapping".

If the goal of the routine is to exclude items, then `.filter()` must be used.
If you then need the preserved items to be mapped to a different type / value, then chain to `.map()`

Reply:
Also, I realize that the term "tab" (for "table" in French, if I am not mistaken) is misused and confusing here. In English I would use "array" or "list" ("table" sounds like a 2D matrix columns x rows)

Reply:
Resolving, as the code has now changed significantly.